submarine_server.py

```diff
@@ -20,13 +24,26 @@ def __init__(self, d=36, y=6, r=6, m=10, L=4, p=2):
         self.red_alert = [i % 100 for i in range(d, d+6)]
         self.submarine_time_left = self.trench_time_left = 120
         self.submarine_location = randint(0, 99)
-        self.is_submarine_in_red = self.submarine_location in self.red_alert
+        print('Waitin on port %s for players...' % PORT)
+        if gui:
+            self.is_submarine_in_red = self.submarine_location in self.red_alert
+            self.web_server = WebsocketServer(WEB_PORT, host=WEB_HOST)
+            self.web_server.new_client = self.front_end_connected
```

Comment:
I'm confused.. what does this do? Is this your way of calling `front_end_connected` if `gui=True`?